ui/components/multichain/pages/review-permissions-page/review-permissions-page.tsx

```diff
@@ -145,28 +165,83 @@ export const ReviewPermissions = () => {
     setShowNetworkToast(true);
   };
 
-  const accounts = useSelector(getUpdatedAndSortedAccounts);
-  const evmAccounts: MergedInternalAccount[] = useMemo(() => {
-    return accounts.filter((account: InternalAccount) =>
-      isEvmAccountType(account.type),
-    );
-  }, [accounts]);
+  const allAccounts = useSelector(
+    getUpdatedAndSortedAccountsWithCaipAccountId,
+  ) as MergedInternalAccountWithCaipAccountId[];
 
-  const connectedAccountAddresses = useSelector((state) =>
-    getPermittedAccountsForSelectedTab(state, activeTabOrigin),
-  ) as string[];
+  const nonRemappedConnectedAccountAddresses = useSelector((state) =>
+    getAllPermittedAccountsForSelectedTab(state, activeTabOrigin),
+  ) as CaipAccountId[];
+
+  // This remaps EVM caip account addresses to match the 'eip155:0'
+  // value that is currently set in InternalAccount.scopes[0] for
+  // EOA EVM accounts. This logic will need to be updated to
+  // support non EOA accounts.
+  const connectedAccountAddresses = uniq(
+    nonRemappedConnectedAccountAddresses.map((caipAccountId) => {
+      const {
+        address,
+        chain: { namespace },
+      } = parseCaipAccountId(caipAccountId);
+      if (namespace === KnownCaipNamespace.Eip155) {
+        // this is very hacky, but it works for now
+        return `eip155:0:${address}` as CaipAccountId;
+      }
+      return caipAccountId;
+    }),
+  );
 
-  const handleSelectAccountAddresses = (addresses: string[]) => {
-    if (addresses.length === 0) {
+  const handleSelectAccountAddresses = (caipAccountIds: CaipAccountId[]) => {
+    if (caipAccountIds.length === 0) {
       setShowDisconnectAllModal(true);
       return;
     }
 
+    const parsedCaipAccountIds = caipAccountIds.map((caipAccountId) => {
+      return parseCaipAccountId(caipAccountId);
+    });
+
+    const addresses = parsedCaipAccountIds.map(({ address }) => address);
+
+    // TODO: we should refactor addPermittedAccounts to accept CaipAccountIds
     dispatch(addPermittedAccounts(activeTabOrigin, addresses));
 
-    connectedAccountAddresses.forEach((address: string) => {
-      if (!addresses.includes(address)) {
-        dispatch(removePermittedAccount(activeTabOrigin, address));
+    connectedAccountAddresses.forEach((connectedAddress: string) => {
+      // TODO: seems like similar logic to selector logic in ui/index.js
+      // See if we can DRY this
+      const parsedConnectedAddress = parseCaipAccountId(
+        connectedAddress as CaipAccountId,
+      );
+
+      const includesCaipAccountId = parsedCaipAccountIds.some(
+        (parsedAddress) => {
+          if (
+            parsedConnectedAddress.chain.namespace !==
+              parsedAddress.chain.namespace ||
+            !isEqualCaseInsensitive(
```

Comment:
Not case sensitive

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/29887/commits/0739f4011195ed1854043ed013f56a589bb179e3